### PR TITLE
RDP Panel: search filter on packets, actors and facctories panels

### PR DESCRIPTION
**Pull request by rpl**
Temporary pullrequest to review the prototype:
- render React components once and propagate state changes using `setState` method
- enable search filter on packet list, actors pools and actors factories panels 
  (currently it simply searches the searchFilter string in the data object converted into a string using `JSON.stringify`)

Prototype preview screencast:

firebugnextrdppanelsearchfilter

### CODE REVIEW

**Comment by rpl** [comment]
@janodvarko I've created this temporary pullrequest to evaluate the changes applied to the RDP Inspector panel by this prototype

File: data/actor-inspector/factories.js [hunk 1]
```diff
@@ -56,22 +62,31 @@ var FactoryTable = React.createClass({
  * TODO docs
  */
 var FactoryList = React.createClass({
+  getInitialState: function() {
+    return {
+      packet: [
+        {factories: {}},
+        {factories: {}}
```
Comment (janodvarko) [inline data/actor-inspector/factories.js]: The 'factories' field is duplicated (should be searchFilter I guess)

Honza

**Comment by janodvarko** [comment]
Looks great!

Please see my comment and I think we can merge then.

Honza

File: data/actor-inspector/factories.js (hunk 1 above)
Comment (rpl) [inline data/actor-inspector/factories.js]: @Honza they are the empty`main` and `child` factories, but it's probably better to if I turn it into a more explicit:

```
return {
  main: {},
  child: {}
}
```

I'm going to update this pull request accordingly

Comment (rpl) [inline data/actor-inspector/factories.js]: @Honza or better: I'm going to update the branch and open this pull request on the firebug/firebug.next repo

**Comment by janodvarko** [comment]
I saw your changes, looks good to me.

> @Honza or better: I'm going to update the branch and open this pull request on the
> firebug/firebug.next repo

Excellent!

Honza

**Comment by honza** [comment]
O HAI, how can I help?

**Comment by rpl** [comment]
my fault, it was a ping to janodvarko
sorry for the noise @honza, I didn't want to flood you ;-)

**Comment by rpl** [comment]
closing this temporary pullrequest, moved to firebug/firebug.next#339

**Pull request closed without merging**